tests/test_tasks.py

```diff
@@ -1,118 +1,137 @@
+    spy.assert_has_calls((max_repetitions + 1) * [mocker.call(seconds)], any_order=True)
 
 
 @pytest.mark.asyncio
-async def test_repeat_print(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.01, max_repetitions=3)
-    async def repeatedly_print_hello() -> None:
-        print("hello")
+async def test_repeat_every_with_sync_function_and_raise_exceptions_false() -> None:
+    seconds = 0.01
+    max_repetitions = 3
+
+    @repeat_every(seconds=seconds, max_repetitions=max_repetitions)
+    def raise_exc() -> NoReturn:
+        raise ValueError("error")
 
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.1)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 3
-    assert err == ""
+    try:
+        await raise_exc()
+    except ValueError as e:
+        pytest.fail(
+            f"{test_repeat_every_with_sync_function_and_raise_exceptions_false.__name__} raised an exception: {e}"
+        )
 
 
 @pytest.mark.asyncio
-async def test_repeat_print_delay(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=3)
-    def repeatedly_print_hello() -> None:
-        print("hello")
+async def test_repeat_every_with_sync_function_and_raise_exceptions_true() -> None:
+    seconds = 0.01
 
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.1)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 2
-    assert err == ""
+    @repeat_every(seconds=seconds, raise_exceptions=True)
+    def raise_exc() -> NoReturn:
+        raise ValueError("error")
+
+    with pytest.raises(ValueError):
+        await raise_exc()
 
 
 @pytest.mark.asyncio
-async def test_repeat_print_wait(capsys: CaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=3, wait_first=0.1)
-    async def repeatedly_print_hello() -> None:
-        print("hello")
+async def test_repeat_every_with_async_function_and_max_repetitions(mocker: MockerFixture) -> None:
+    counter = 0
+    seconds = 0.01
+    max_repetitions = 3
+    spy = mocker.spy(asyncio, "sleep")
+
+    @repeat_every(seconds=seconds, max_repetitions=max_repetitions)
+    async def increase_counter() -> None:
+        nonlocal counter
+        counter += 1
 
-    await repeatedly_print_hello()
-    await asyncio.sleep(0.15)
-    out, err = capsys.readouterr()
-    assert out == "hello\n" * 1
-    assert err == ""
+    await increase_counter()
+
+    assert counter == max_repetitions
+    spy.assert_has_calls(max_repetitions * [mocker.call(seconds)], any_order=True)
 
 
 @pytest.mark.asyncio
-async def test_repeat_unlogged_error(caplog: LogCaptureFixture) -> None:
-    @repeat_every(seconds=0.07, max_repetitions=None)
-    def log_exc() -> NoReturn:
-        raise ValueError("repeat")
+async def test_repeat_every_with_async_function_max_repetitions_and_wait_first(mocker: MockerFixture) -> None:
+    counter = 0
+    seconds = 0.01
+    max_repetitions = 3
+    spy = mocker.spy(asyncio, "sleep")

```

Comment:
sleep can be consted I think...

Reply:
I think it's unnecessary, it's the name of the function contained in the `asyncio` module. The developers should know this function to understand the tests (and be good developers ;))